Cloudflare integration announcement (#174)
xataio/mdx-blog: cloudflare-integration.mdx

---
title: 'Connect your Cloudflare Worker with Xata in one click'
description: 'Connecting a Cloudflare Worker with Xata is now possible from the Cloudflare dashboard.'
image:

>   src: https://raw.githubusercontent.com/xataio/mdx-blog/main/images/xata-workers.png

Author: ```suggestion
  src: https://raw.githubusercontent.com/xataio/mdx-blog/main/images/xata-cloudflare.png
```

  alt: Xata + Cloudflare
author: Alexis Rico

> date: 12-12-2023

Author: ```suggestion
date: 12-14-2023
```

tags: ['cloudflare', 'serverless', 'database', 'integration']

> published: true

Author: ```suggestion
published: false
```

slug: cloudflare-integration
---

> Cloudflare Workers, a serverless platform, empowers developers to run code at the edge of the network. This means running code closer to users, resulting in faster response times and lower latency. Cloudflare Workers also offer a secure environment, ensuring code isolation from other users on the same server.

Reviewer: ```suggestion
Cloudflare Workers is a serverless platform that empowers developers to run code at the edge of the network. This means running code closer to users, resulting in faster response times and lower latency. Cloudflare Workers also offers a secure environment, ensuring code isolation from other users on the same server.
```

> Cloudflare Workers is a serverless platform that empowers developers to run code at the edge of the network. This means running code closer to users, resulting in faster response times and lower latency. Cloudflare Workers also offers a secure environment, ensuring code isolation from other users on the same server.

Reviewer 2: ```suggestion
[Cloudflare Workers](https://developers.cloudflare.com/workers/) is a serverless platform that empowers developers to run code at the edge of the network. This means running code closer to users, resulting in faster response times and lower latency. Cloudflare Workers also offers a secure environment, ensuring code isolation from other users on the same server.
```

We are thrilled to announce a collaboration with Cloudflare, enabling a one-click integration directly from the Cloudflare dashboard. This user-friendly solution eliminates the need for developers to create an API key and manually configure their database connection.

<ArticleImage

>   src="https://raw.githubusercontent.com/xataio/mdx-blog/cloudflare/images/cloudflare-integration-dashboard.png"

Author: ```suggestion
  src="https://raw.githubusercontent.com/xataio/mdx-blog/main/images/cloudflare-integration-dashboard.png"
```

  width="874"
  height="509"
  alt="Cloudflare Integration Image"
  caption="Cloudflare dashboard with the Xata integration"
  position="center"
/>

> Simply visit the Cloudflare dashboard, navigate to the "Integrations" tab, and find the Xata integration. Click on the "Add integration" button, and a wizard will guide you through the process of connecting your Cloudflare Worker with Xata.

Reviewer 2: This is a small nit, but I believe the integrations tab needs to enabled. I'm not sure if it's there by default.

https://developers.cloudflare.com/workers/databases/native-integrations/

<ArticleImage
  src="https://raw.githubusercontent.com/xataio/mdx-blog/cloudflare/images/cloudflare-integration-setup.png"
  width="456"
  height="558"
  alt="Cloudflare Integration Image"
  caption="Wizard to connect your Cloudflare Worker with Xata"
  position="center"
/>

During the process, you'll be prompted to authorize Cloudflare to access your Xata account. Once authorized, you can select the database you want to connect to your worker.

<ArticleImage

>   src="https://raw.githubusercontent.com/xataio/mdx-blog/cloudflare/images/cloudflare-integration-authorization.png"

Author: ```suggestion
  src="https://raw.githubusercontent.com/xataio/mdx-blog/main/images/cloudflare-integration-authorization.png"
```

  width="580"
  height="613"
  alt="Cloudflare Integration Image"
  caption="Page to authorize Cloudflare to access your Xata account"
  position="center"
/>

After authorizing Cloudflare, you'll be redirected back to the Cloudflare dashboard, where you can choose the database to connect to your worker.

<ArticleImage

>   src="https://raw.githubusercontent.com/xataio/mdx-blog/cloudflare/images/cloudflare-integration-selection.png"

Author: ```suggestion
  src="https://raw.githubusercontent.com/xataio/mdx-blog/main/images/cloudflare-integration-selection.png"
```

  width="810"
  height="843"
  alt="Cloudflare Integration Image"
  caption="Page to select the database you want to connect to your worker"
  position="center"
/>

All necessary environment variables will be automatically added to your worker.

<ArticleImage

>   src="https://raw.githubusercontent.com/xataio/mdx-blog/cloudflare/images/cloudflare-integration-secrets.png"

Author: ```suggestion
  src="https://raw.githubusercontent.com/xataio/mdx-blog/main/images/cloudflare-integration-secrets.png"
```

  caption="List of environment variables added to your worker"
  position="center"
/>

In your worker, use the environment variables to connect to your database.

```ts
import { XataClient } from './xata';

export interface Env {
  XATA_BRANCH: string;
  XATA_API_KEY: string;
  XATA_DATABASE_URL: string;
}

export default {
  async fetch(request: Request, env: Env, ctx: ExecutionContext): Promise<Response> {
    const xata = new XataClient({
      apiKey: env.XATA_API_KEY,
      branch: env.XATA_BRANCH,
      databaseURL: env.XATA_DATABASE_URL
    });

    const posts = await xata.db.Posts.getAll();
    return new Response(`Total Posts: ${posts.length}`);
  }
};
```

## How does it work?

The integration is powered by an OAuth2 flow, allowing Cloudflare to access your Xata account and create a new API key for your worker. The API key is then used to connect your worker with your database.

For more details on integrating with Xata using OAuth2, refer to our [documentation](/docs/integrations/oauth).

## What's next?

The integration is now available for all Cloudflare users. To learn more about how to use it, consult our [documentation](/docs/integrations/cloudflare).

> For questions or feedback, feel free to reach out to us on [Twitter](https://twitter.com/xata) or [Discord](https://xata.io/discord).

Reviewer 2: ```suggestion
For questions or feedback, feel free to reach out to us on [X / Twitter](https://twitter.com/xata) or [Discord](https://xata.io/discord). [Sign up](https://app.xata.io/signin?mode=signup) today to get started with Xata 🚀 
```

---
title: 'Xata + Cloudflare: One Click Database Integration with a Worker'
description: 'Connecting a Cloudflare Worker with Xata is now possible from the Cloudflare dashboard.'
image:

>   src: https://raw.githubusercontent.com/xataio/mdx-blog/main/images/xata-workers.png

Author: TODO

Author: @alexfrancoeur We need an illustration

  alt: Xata + Cloudflare
author: Alexis Rico

> date: 12-12-2023

Author: Update

## What is Cloudflare Workers?

Cloudflare Workers is a serverless platform that allows developers to run code at the edge of the network. This means that you can run code closer to your users, which results in faster response times and lower latency. Cloudflare Workers also provide a secure environment for running code, so you don't have to worry about isolating your code from other users on the same server.

## How does the integration look like?

The integration is simple and straightforward, visit the Cloudflare dashboard and inside the "Integrations" tab, you will find the Xata integration. Click on the "Add integration" button and you will be redirected to the Xata dashboard where you can create a new database or select an existing one. Once you have selected the database, you will be redirected back to the Cloudflare dashboard and all the necessary environment variables will be automatically added to your worker.

<ArticleImage

>   src="https://raw.githubusercontent.com/xataio/mdx-blog/cloudflare/images/cloudflare-integration-setup.png"

Author: ```suggestion
  src="https://raw.githubusercontent.com/xataio/mdx-blog/main/images/cloudflare-integration-setup.png"
```
